### PR TITLE
Reload manager defaults at daemon-reload

### PR DESCRIPTION
**Pull request by tblume**
"systemctl daemon-reload" should also update the manager defaults from
/etc/systemd/system.conf.
For details, see:
http://lists.freedesktop.org/archives/systemd-devel/2015-June/033062.html

### CODE REVIEW

File: src/core/main.c [hunk 1]
```diff
@@ -685,6 +685,29 @@ static int parse_config_file(void) {
...
+        m->default_restart_usec = arg_default_restart_usec;
+        m->default_start_limit_interval = arg_default_start_limit_interval;
+        m->default_start_limit_burst = arg_default_start_limit_burst;
+
+        manager_set_default_rlimits(m, arg_default_rlimit);
+        manager_environment_add(m, NULL, arg_default_environment);
+
+        return 0;
+}
+
```
Comment (zonque) [inline src/core/main.c:707]: Could you rename this function to `manager_set_default()` or something and then also call it from `main()` to get rid of the code duplication?

**Comment by tblume** [comment]
Thanks for the hint, I have amended the commit.

File: src/core/main.c [hunk 2]
```diff
@@ -685,6 +685,33 @@ static int parse_config_file(void) {
         return 0;
 }
 
+static int manager_set_defaults(Manager *m) {
+        int r;
+
+        assert(m);
+
+        r = parse_config_file();
```
Comment (zonque) [inline src/core/main.c]: Now `parse_config_file()` is invoked twice. I'd suggest to remove it here, make the function `void` and …

File: src/core/main.c [hunk 3]
```diff
@@ -1763,6 +1777,11 @@ int main(int argc, char *argv[]) {
 
                 case MANAGER_RELOAD:
                         log_info("Reloading.");
+
```
Comment (zonque) [inline src/core/main.c:1773]: … add move the call to here.

**Comment by zonque** [comment]
Jup, looks good to me now, thanks!

**Comment by zonque** [comment]
Ah, except your pull-request branch is now convoluted. Could you rebase it against current master please?

**Comment by zonque** [comment]
The result should be just one commit in the branch.

**Comment by tblume** [comment]
Sorry, I've trashed my branch by some rebase attempts.
Hope the commit is ok now.

**Comment by zonque** [comment]
Yes, thanks!

**Pull request merged**